### DISCUSSION TITLE
Error during runtime after creating a trigger: record "new" has no field

### DISCUSSION THREAD

**Discussion by sagarvora** [supabase/supabase#2451] [Questions] [answered]
I've created the following trigger from a Query:

```sql
create or replace function set_is_active()
returns trigger as $$
begin
  update airports set is_active = true WHERE identifier = new.home_airport;
  return new;
end;

$$ language plpgsql security definer;
drop trigger if exists home_airport_changed on profiles;
create trigger home_airport_changed
after insert or update on profiles
for each row execute procedure set_is_active();
```

The trigger gets created successfully. For some reason when I try to update a row, I get the following error:
```sql
record "new" has no field "home_airport"

```

What am I doing wrong?

Thanks.

**EDIT: (additional info)**
If I change `new.home_airport` to any other text column (e.g. `new.last_name`), I am able to get it to work. This issue is occurring only for columns which are also foreign keys (`home_airport` and `id`)

Also, adding something like this to the function works as expected:
```
raise exception 'ERROR: %', new.home_airport;
```

**Comment by burggraf** [comment]
I think you want to use `BEFORE UPDATE` and not `AFTER UPDATE` for your trigger.

[See this post](https://dba.stackexchange.com/questions/153525/what-is-the-difference-between-an-after-update-and-a-before-update-in-postgresql)

**Reply by sagarvora** [reply]
Thanks for responding. Unfortunately, this didn't fix the issue for me.

I've updated the question with more info about this.

**Reply by burggraf** [reply]
ah -- `This issue is occurring only for columns which are also foreign keys` .  This has to be the key.
Is [this](https://dba.stackexchange.com/questions/164474/foreign-key-violation-within-trigger-on-insert-of-new-id-in-other-table-as-for) helpful?  I don't think I've tried to write a trigger that updates a field that's a foreign key before.

**Comment by sagarvora** [comment, marked as answer]
I've solved this issue. Turned out there were some incorrect triggers on the `airports` table (the one I was trying to update in above trigger) that were causing this.

**Reply by burggraf** [reply]
Thank you for following up here!

**Reply by sagarvora** [reply]
Thank _you_ for all the assistance! I really appreciate it.
